plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/model/CodeWhispererModel.kt

```diff
@@ -81,10 +93,19 @@ data class SupplementalContextInfo(
 }
 
 data class RecommendationContext(
-    val details: List<DetailContext>,
+    val details: MutableList<DetailContext>,
```

Comment:
not seeing why need mutable

Reply:
Now each trigger will only have one state `InvocationContext` and is stored in `ongoingRequests`, when subsequent suggestions sent back to client, they will be added to this mutable details, and other parts of `InvocationContext` remains the same.

```
private fun updateStates(
        states: InvocationContext,
        response: GenerateCompletionsResponse
    ): InvocationContext {
        val recommendationContext = states.recommendationContext
        val newDetailContexts = CodeWhispererRecommendationManager.getInstance().buildDetailContext(
            states.requestContext,
            recommendationContext.userInputSinceInvocation,
            response.completions(),
            response.responseMetadata().requestId()
        )

        recommendationContext.details.addAll(newDetailContexts)
        return states
    }
```